includes/wikia/controllers/HeliosHelperController.class.php

```diff
@@ -151,31 +173,86 @@ public function sendTemporaryPasswordEmail() {
 		if ( !isset( $tempPassword ) ) {
 			return;
 		}
-
 		$user = \User::newFromName( $username );
 
-		if ( ! $user instanceof \User ) {
-			$this->response->setVal( 'message', 'unable to create a \User object from name' );
+		if ( !$user instanceof \User ) {
+			$this->response->setVal( self::FIELD_MESSAGE, 'unable to create a \User object from name' );
 			$this->response->setCode( \WikiaResponse::RESPONSE_CODE_BAD_REQUEST );
+
 			return;
 		}
 
-		if ( ! $user->getId() ) {
-			$this->response->setVal( 'message', 'no such user' );
+		if ( !$user->getId() ) {
+			$this->response->setVal( self::FIELD_MESSAGE, 'no such user' );
 			$this->response->setCode( \WikiaResponse::RESPONSE_CODE_BAD_REQUEST );
+
 			return;
 		}
 
 		$resp = \F::app()->sendRequest( 'Email\Controller\ForgotPassword', 'handle', [
 			'targetUser' => $username,
-			'tempPass' => $tempPassword,
+			'tempPass'   => $tempPassword,
 		] );
 
 		$data = $resp->getData();
-		if ( !empty( $data['result'] ) && $data['result'] == 'ok' ) {
-			$this->response->setVal( 'success', true );
+		if ( !empty( $data[ self::FIELD_RESULT ] ) && $data[ self::FIELD_RESULT ] == 'ok' ) {
+			$this->response->setVal( self::FIELD_SUCCESS, true );
 		} else {
-			$this->response->setVal( 'message', 'could not send an email message' );
+			$this->response->setVal( self::FIELD_MESSAGE, self::ERR_COULD_NOT_SEND_AN_EMAIL_MESSAGE );
+			$this->response->setCode( \WikiaResponse::RESPONSE_CODE_NOT_FOUND );
+		}
+	}
+
+	/**
+	 * Api endpoint to send a password reset e-mail.
+	 * @requestParam user_id : \User::id
+	 * @requestParam token : user identification token
+	 */
+	public function sendPasswordResetLinkEmail() {
+		$this->response->setFormat( 'json' );
+		$this->response->setCacheValidity( \WikiaResponse::CACHE_DISABLED );
+		$this->response->setVal( self::FIELD_SUCCESS, false );
+
+		if ( !$this->authenticateViaTheSchwartz() ) {
+			return;
+		}
+
+		$userId = $this->getFieldFromRequest( 'user_id', self::ERR_INVALID_USER_ID );
+		$token = $this->getFieldFromRequest( self::FIELD_TOKEN, self::ERR_INVALID_TOKEN );
+		$returnUrl = $this->getVal( 'return_url', null );
+
+		if ( empty( $userId ) || empty ( $token ) ) {
+			return;
+		}
+
+		wfWaitForSlaves( $this->wg->ExternalSharedDB );
```

Comment:
It was needed in case of the account confirmation method, but in case of password reset token I think it is obsolete.